mmengine/optim/scheduler/param_scheduler.py

```diff
@@ -1283,3 +1283,203 @@ def get_position_from_periods(
             if iteration < period:
                 return i
         return None
+
+
+@PARAM_SCHEDULERS.register_module()
+class ReduceOnPlateauParamScheduler(_ParamScheduler):
+    """ReduceOnPlateauParamScheduler."""
+    _default_monitor_stages = ['val']
+
+    def __init__(
+            self,
+            optimizer: OptimizerType,
+            param_name: str,
+            monitor: str = 'val.loss',  # monitor_stage.monitor_key
+            monitor_stages: Optional[Sequence[str]] = None,
+            rule: str = 'min',
+            factor: float = 0.1,
+            patience: int = 10,
+            threshold: float = 1e-4,
+            threshold_rule: str = 'rel',
+            cooldown: int = 0,
+            min_value: float = 0,
+            eps: float = 1e-8,
+            begin: int = 0,
+            end: int = INF,
+            last_step: int = -1,
+            by_epoch: bool = True,
+            verbose: bool = False):
+
+        # Attach optimizer
+        if not isinstance(optimizer, (Optimizer, OptimWrapper)):
```

Comment:
Why don't we call `super().__init__` here?

Reply:
Because we don't need the weakref for the optimizer to build with_count function.